src/duplicacy_gcdstorage.go

```diff
@@ -83,18 +93,37 @@ func (storage *GCDStorage) shouldRetry(threadIndex int, err error) (bool, error)
 		retry = err.Temporary()
 	}
 
-	if !retry || storage.backoffs[threadIndex] >= 256 {
-		storage.backoffs[threadIndex] = 1
+	if !retry || storage.backoffsRetries[threadIndex] >= MAX_NUMBER_OF_RETRIES {
+		LOG_INFO("GCD_RETRY", "Thread: %03d. Maximum number of retries reached. Backoff time: %.2f. Number of retries: %d", threadIndex, storage.backoffs[threadIndex], storage.backoffsRetries[threadIndex])
+		storage.backoffs[threadIndex] = computeInitialBackoff(minimumSleep, maximumSleep)
+		storage.backoffsRetries[threadIndex] = 0
 		return false, err
 	}
 
-	delay := float32(storage.backoffs[threadIndex]) * rand.Float32()
-	LOG_DEBUG("GCD_RETRY", "%s; retrying after %.2f seconds", message, delay)
-	time.Sleep(time.Duration(float32(storage.backoffs[threadIndex]) * float32(time.Second)))
-	storage.backoffs[threadIndex] *= 2
+	if storage.backoffs[threadIndex] < LIMIT_BACKOFF_TIME {
```

Comment:
This logic is going to result in you going over your limit, which isn't the goal. You also have a bug where you're only incrementing the retries count when you're over the limit.

Instead, do the following:

```storage.backoffs[threadIndex] *= 2
if storage.backoffs[threadIndex] > LIMIT_BACKOFF_TIME {
    storage.backoffs[threadIndex] = LIMIT_BACKOFF_TIME
}
storage.backoffsRetries[threadIndex]++
```